lib/gui/modules/drive-scanner.js

```diff
@@ -16,86 +16,130 @@
-)
+/**
+ * @summary Time to delay the first scan (in milliseconds)
+ * @description This prevents the drive scans from kicking
+ * off while Etcher is still loading, which would
+ * considerably delay the time until the main window shows
+ * @type {Number}
+ * @constant
+ */
+const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 125
 
-/* eslint-enable lodash/prefer-lodash-method */
+/**
+ * @summary DriveScanner class
+ * @class
+ * @extends {EventEmitter}
+ */
+class DriveScanner extends EventEmitter {
+  /**
+   * @summary DriveScanner constructor
+   * @example
+   * const scanner = new DriveScanner()
+   *   .on('error', (error) => { ... })
+   *   .on('drives', (drives) => { ... })
+   */
+  constructor () {
+    super()
+    this.isRunning = false
+    this.timer = null
+    this.errorCount = 0
+    this.run = this.run.bind(this)
+  }
 
-  .flatMap(() => {
-    return Rx.Observable.fromNodeCallback(drivelist.list)()
-  })
+  /**
+   * @summary Start the scanner
+   * @public
+   * @example
+   * scanner.start()
+   */
+  start () {
+    debug('start')
+    if (this.isRunning) {
+      return
+    }
+    this.isRunning = true
+    this.timer = setTimeout(this.run, DRIVE_SCANNER_FIRST_SCAN_DELAY_MS)
+  }
 
-  .map((drives) => {
-    if (settings.get('unsafeMode')) {
-      return drives
+  /**
+   * @summary Kick off a scan & schedule the next run
+   * @private
+   * @example
+   * scanner.run()
+   */
+  run () {
+    if (!this.isRunning) {
+      return
     }
 
-    return _.reject(drives, {
-      system: true
+    DriveScanner.scan((error, drives) => {
+      debug('scan', error || drives)
+      if (error) {
+        this.errorCount += 1
+        this.emit('error', error)
+      } else {
+        this.emit('drives', drives)
+      }
+      if (this.isRunning) {
+        this.timer = setTimeout(this.run, DRIVE_SCANNER_DELAY_MS)
+      }
     })
-  })
-  .pausable(new Rx.Subject())
+  }
 
-/*
- * This service emits the following events:
- *
- * - `drives (Object[])`
- * - `error (Error)`
- *
- * For example:
- *
- * ```
- * driveScanner.on('drives', (drives) => {
- *   console.log(drives);
- * });
- *
- * driveScanner.on('error', (error) => {
- *   throw error;
- * });
- * ```
- */
-availableDrives.subscribe((drives) => {
-  emitter.emit('drives', drives)
-}, (error) => {
-  emitter.emit('error', error)
-})
+  /**
+   * @summary Stop scanning for drives
+   * @public
+   * @example
+   * scanner.stop()
+   */
+  stop () {
+    debug('stop')
+    this.isRunning = false
+    clearTimeout(this.timer)
+  }
 
-/**
- * @summary Start scanning drives
- * @function
- * @public
- *
- * @example
- * driveScanner.start();
- */
-emitter.start = () => {
-  availableDrives.resume()
-}
+  /**
+   * @summary Scan for drives, optionally filtering out system drives
+   * @private
+   * @static
+   * @param {Function} callback - callback(error, drives)
+   * @example
+   * DriveScanner.scan((error, drives) => {
+   *   ...
+   * })
+   */
+  static scan (callback) {
```

Comment:
I was initially a bit confused about there being no equivalent of the `clearTimeout` in the code above, but then I had a more careful look at the code, and realised that even Jonas's current solution would still work if the `clearTimeout` was removed ;-)